File: web/di_core/LakeHouse/Domain/LakeJob/LakeJobResponse.ts
import { LakeJob } from '@core/LakeHouse/Domain/LakeJob/LakeJob';
import { UserProfile } from '@core/domain';

export class LakeJobResponse {
  job: LakeJob;
  creator: UserProfile;

  constructor(job: LakeJob, creator: UserProfile) {
    this.job = job;
    this.creator = creator;
  }

  static fromObject(obj: any): LakeJobResponse {
    const job = LakeJob.fromObject(obj.job);
    const creator = UserProfile.fromObject(obj.creator);
    return new LakeJobResponse(job, creator);
  }
}

File: job-scheduler/src/main/scala/datainsider/lakescheduler/domain/response/LakeJobResponse.scala
package datainsider.lakescheduler.domain.response

import datainsider.client.domain.user.ShortUserProfile
import datainsider.jobscheduler.domain.Ids.RunId
import datainsider.lakescheduler.domain.job.LakeJob

case class NextLakeJobResponse(hasJob: Boolean, data: Option[LakeRunInfo])

case class LakeJobResponse(job: LakeJob, creator: Option[ShortUserProfile])

case class LakeRunInfo(runId: RunId, job: LakeJob)
